plonk-core/src/constraint_system/arithmetic.rs

```diff
@@ -136,7 +136,7 @@ where
         self.q_lookup.push(F::zero());
 
         if let Some(pi) = gate.pi {
-            self.public_inputs.insert(self.n, pi);
+            self.add_pi(self.n, &pi).unwrap();
```

Comment:
The 2 scenarios in which a panic would occur are:
 1. `pi` can not be converted to a field element, which will never happen as it already is a field element.
 2. `self.n` position is already filled with a value, which could occur in the case that something went very wrong when keeping track of the current row in the circuit; something like having a mistake in the number of rows a gate uses.

Reply:
Then it would make sense to panic. 

But let's use `expect` with some message that hints why the error is not expected rather than simply use `unwrap`.

Reply:
Added `unwrap_or_else` with error message :+1: